kbase-extension/static/kbase/js/common/cellComponents/cellTabs.js

```diff
@@ -72,22 +74,26 @@ define(['bluebird', 'common/html', 'common/ui', 'common/events'], (Promise, html
 
         function buildTabButtons(events) {
             const buttons = Object.keys(controlBarTabs.tabs)
+                .filter((key) => {
+                    // ensure that the tab data is an object with key 'label'
+                    return (
+                        controlBarTabs.tabs[key] &&
+                        typeof controlBarTabs.tabs[key] === 'object' &&
+                        controlBarTabs.tabs[key].label
+                    );
+                })
                 .map((key) => {
                     const tab = controlBarTabs.tabs[key];
-                    let icon;
                     if (!tab) {
                         console.warn('Tab not defined: ' + key);
                         return;
                     }
-                    if (tab.icon) {
-                        if (typeof tab.icon === 'string') {
-                            icon = {
-                                name: tab.icon,
-                                size: 2,
-                            };
-                        } else {
-                            icon = { size: 2 };
-                        }
+                    let icon;
+                    if (tab.icon && typeof tab.icon === 'string') {
```

Comment:
ensure there is an icon name, as otherwise the UI will reject icon-building attempts